### PR TITLE
Make ops classes final and extend AnyVal

### PR DESCRIPTION
**Pull request by edmundnoble**
I am not sure how much this will matter in dotty, but this should minimize allocations and be more friendly to the inliner.

### CODE REVIEW

**Comment by etorreborre** [comment]
The compilation failed for the `monix` and `scalaz` modules, can you please have a look at that? Otherwise I think this is a good idea :-).

**Comment by edmundnoble** [comment]
I believe there's some intermittent failures in monix and scalaz task tests affecting this build.

**Comment by etorreborre** [comment]
Can you rebase with master because I did some changes today to stabilize the build? Thanks.

**Comment by edmundnoble** [comment]
Rebased.

File: monix/shared/src/main/scala/org/atnos/eff/monix/AsyncTaskService.scala [hunk 1]
```diff
@@ -40,7 +40,7 @@ case class AsyncTaskService() extends AsyncService {
 trait AsyncTaskServiceInterpretation {
 
   def runAsyncTask[A](e: Eff[Fx.fx1[Async], A]): Task[A] =
-    e.detachA(ApplicativeAsync) match {
```
Comment (edmundnoble) [inline monix/shared/src/main/scala/org/atnos/eff/monix/AsyncTaskService.scala]: Ohhhh my, I see. The monad is getting picked by implicit resolution. Gosh darn it :P

**Comment by etorreborre** [comment]
Thanks!

**Pull request merged**